Lib/fontTools/pens/basePen.py

```diff
@@ -141,16 +142,52 @@ def addComponent(self, glyphName, transformation):
 		pass
 
 
-class BasePen(AbstractPen):
+class DecomposingPenMixin(LogMixin):
+	""" Pen mixin class implementing a 'addComponent' method that decomposes
+	components (i.e. draws them onto self as simple contours).
+
+	The constructor takes a single 'glyphSet' positional argument, a dict
+	where the glyphs referenced as components are looked up by their name.
+
+	Subclasses can set the 'skipMissingComponents' attribute to True, if they
+	want to ignore the 'KeyError' exception normally raised when a base glyph
+	is missing from the glyphSet, and replace it by a warning message.
+	"""
+
+	skipMissingComponents = False
+
+	def __init__(self, glyphSet):
+		super(DecomposingPenMixin, self).__init__()
```

Comment:
the call to `super` here is ok if the `__init__` method of the next class in the inheritance line does not take any required arguments (as is the case of the `RecordingPen`).
However, it breaks if the super class constructor requires any arguments... Maybe that's ok.